torch/csrc/inductor/aoti_torch/tensor_converter.cpp

```diff
@@ -0,0 +1,56 @@
+
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+#include <torch/csrc/inductor/aoti_torch/tensor_converter.h>
+
+namespace torch {
+namespace aot_inductor {
+
+at::Tensor* tensor_handle_to_tensor_pointer(AtenTensorHandle handle) {
+  return reinterpret_cast<at::Tensor*>(handle);
+}
+
+AtenTensorHandle tensor_pointer_to_tensor_handle(at::Tensor* tensor) {
+  return reinterpret_cast<AtenTensorHandle>(tensor);
+}
+
+std::vector<AtenTensorHandle> borrow_tensors_to_handles(
+    std::vector<at::Tensor>& tensors) {
+  std::vector<AtenTensorHandle> result;
+  result.reserve(tensors.size());
+  for (auto tensor : tensors) {
+    auto allocated = new at::Tensor(std::move(tensor));
```

Comment:
I am going with borrow only in this version, safe to start with. If we do steal later, we can create another RAII class to make sure the stealing and returning happen in pair.

Reply:
This is not a borrow lol. The result has owning handles and you MUST deallocate them when you are done.

Reply:
I am not sure I entirely understand this comment, but if when you pass a pointer across the ABI, the receiver is expected to free it, of course this is part of the API and you have to consistently expect this. I also think that stealing across the call is the right thing to do (as it allows AOTInductor model.so to promptly deallocate inputs.)

Reply:
Is perhaps the confusion here when you have a shared pointer, sometimes the act of incrementing the refcount is referred to as a "borrow" (I'm specifically thinking of `reinterpret_borrow` in pybind11, which lets you coerce a handle into an object and could not be implemented any other way). We should NOT use this naming convention here because we're dealing with two levels of ownership (the outer, unique exclusive ownership via new/delete, and the inner shared pointer count on at::Tensor) and, as far as the ABI is concerned, only the outer ownership matters. And in any case, when referencing CPython documentation, borrow is always used to mean "did not increment reference count).